### PR TITLE
Add fake applications endpoint

### PR DESCRIPTION
**Pull request by kjdchapman**
TODO:
- [x] use flat files for the applications
- [x] automagically get `rails_helper` into api spec files
- [x] discuss how much testing this needs for the "faked" implementation
- [x] other idiomatic Rails API things I might have missed

(NB: this was branched off https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training/pull/31)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,19 @@
+.PHONY: docker-down
```
Comment (heidar) [inline Makefile]: Is this meant to be in here?

File: app/controllers/api/applications_controller.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+class Api::ApplicationsController < ActionController::API
+  def index
+    render json: [
```
Comment (heidar) [inline app/controllers/api/applications_controller.rb]: Can we start out with returning an object that has a key `applications` that points to an array? Just so we're backwards compatible. Happy to also leave it out at this point and add it later. :)

File: spec/api/viewing_all_applications_spec.rb [hunk 3]
```diff
@@ -0,0 +1,34 @@
+# TODO: import this automagically for /api folder
```
Comment (heidar) [inline spec/api/viewing_all_applications_spec.rb]: Can we put this in `spec/requests/api` as it is a request spec?

File: spec/api/viewing_all_applications_spec.rb [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    headers = { "ACCEPT" => "application/json" }
+    get '/api/applications', headers: headers
+  end
+
+  it 'responds with a success code' do
+    expect(response).to have_http_status(:ok)
+  end
+
+  it 'contains three applications' do
+    expect(JSON.parse(response.body).count).to eq(3)
```
Comment (heidar) [inline spec/api/viewing_all_applications_spec.rb]: Can we DRY `JSON.parse(response.body)`? 🚀

File: Makefile (hunk 1 above)
Comment (kjdchapman) [reply Makefile]: Made Tech's (admittedly emerging) [Makefile standards](https://github.com/madetech/rfcs/blob/master/rfc-012-makefile-standards.md) recommended it.

These stop folders of this name getting confused with command by `make`. Often ran into it with `make test`, and though it's unlikely for these other folder names, I tend to just chuck it on 'em all. 

I _think_ you can do a one-liner to cover all the PHONYs, which might help if this hurts the readability, but I don't think this file is long enough to worry too much.

File: app/controllers/api/applications_controller.rb (hunk 2 above)
Comment (kjdchapman) [reply app/controllers/api/applications_controller.rb]: I actually started with this, then noticed [our API spec](https://editor.swagger.io/?url=https://raw.githubusercontent.com/DFE-Digital/apply-for-postgraduate-teacher-training-api/master/api_spec.yml) did it the other way! 

I think we should change both.

Comment (heidar) [reply app/controllers/api/applications_controller.rb]: Ah, yes, agreed!

File: spec/api/viewing_all_applications_spec.rb (hunk 3 above)
Comment (kjdchapman) [reply spec/api/viewing_all_applications_spec.rb]: Addressed in https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training/pull/33/commits/02d0bb60a7ae90c7b05e910b2759b155903db2e3

File: Makefile (hunk 1 above)
Comment (heidar) [reply Makefile]: Ahh, I see. I thought it was meant to be in the other PR, not in this one.

**Comment by heidar** [comment]
Also sorry, just realised this is a WIP draft, derp.

**Comment by kjdchapman** [comment]
@heidar Not at all, your comments were useful! :tada: 

Any thoughts on where to stick flat json files? I'm not sure what the Rails convention would be for that, or where they'd end up. Asset pipeline?

File: Makefile (hunk 1 above)
Comment (kjdchapman) [reply Makefile]: Oh, right, yeah! This is branched off the Makefile PR, it won't show up in the final version of this'un.

File: app/controllers/api/applications_controller.rb (hunk 2 above)
Comment (kjdchapman) [reply app/controllers/api/applications_controller.rb]: Actually, I say that, and we might have already put this spec in front of people, so changing it from under them might be worth thinking quickly about. 

@himal-mandalia - who have we sent [our swagger link](https://editor.swagger.io/?url=https://raw.githubusercontent.com/DFE-Digital/apply-for-postgraduate-teacher-training-api/master/api_spec.yml) to?

(we'll get it changed, it's just a question of whether _right now_ is correct)

File: spec/api/viewing_all_applications_spec.rb (hunk 4 above)
Comment (kjdchapman) [reply spec/api/viewing_all_applications_spec.rb]: Good shout - addressed in https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training/pull/33/commits/fd576b45957966998d50404349ef6a0f10aaabdc

**Comment by heidar** [comment]
@kjdchapman throw them in `lib/data` maybe? There isn't a convention for this as far as I know

File: app/controllers/api/applications_controller.rb (hunk 2 above)
Comment (himal-mandalia) [reply app/controllers/api/applications_controller.rb]: @kjdchapman we've sent this out to three vendors so we shouldn't be changing it yet - some of them may not have had a chance to look at it yet.

**Comment by himal-mandalia** [comment]
@kjdchapman @heidar there isn't a Rails convention for this but `lib/data` is what I've seen used before.

File: app/controllers/api/applications_controller.rb (hunk 2 above)
Comment (kjdchapman) [reply app/controllers/api/applications_controller.rb]: Added this as [an issue on this repo](https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training/issues/37), and a [related one on the API spec repo](https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training-api/issues/3), so we don't lose sight of it.

**Comment by kjdchapman** [comment]
`lib/data` sounds good to me.

File: app/controllers/api/applications_controller.rb (hunk 2 above)
Comment (himal-mandalia) [reply app/controllers/api/applications_controller.rb]: We might also want to look into HATEOS style payload with links.

**Comment by kjdchapman** [comment]
Interested to hear opinions on whether this is a suitable level of testing.

I could mess up this JSON file and the tests wouldn't fail, but is listing every field worth it? Maybe.

**Comment by himal-mandalia** [comment]
Listing every field could be worth it... but not yet, this is going to be fluid for a while. I'd like to get to an agreed set of fields then lock them in with a spec.

**Pull request merged**